packages/core/src/callbacks/utility/handleOpenAIStream.ts

```diff
@@ -1,64 +1,36 @@
+import { ChatCompletionChunk } from "openai/resources/chat";
 import { globalsHelper } from "../../GlobalsHelper";
 import { StreamCallbackResponse, Event } from "../CallbackManager";
-import { StreamToken } from "../CallbackManager";
+import { APIResponse } from "openai/core";
+import { Stream } from "openai/streaming";
 
 export async function handleOpenAIStream({
   response,
   onLLMStream,
   parentEvent,
 }: {
-  response: any;
+  response: APIResponse<Stream<ChatCompletionChunk>>;
   onLLMStream: (data: StreamCallbackResponse) => void;
   parentEvent?: Event;
 }): Promise<string> {
   const event = globalsHelper.createEvent({
     parentEvent,
     type: "llmPredict",
   });
-  const stream = __astreamCompletion(response.data as any);
   let index = 0;
   let cumulativeText = "";
-  for await (const message of stream) {
-    const token: StreamToken = JSON.parse(message);
-    const { content = "", role = "assistant" } = token?.choices[0]?.delta ?? {};
+  for await (const part of response) {
+    const { content = "", role = "assistant" } = part.choices[0].delta;
```

Comment:
Nit: Shouldn't we be parsing the role out of the response? It could be a function or assistant no? 👀

Reply:
Yeah good point. Right now I think it always returns assistant (even for function calls, where it just uses the function_call payload) but I might as well not hard code assistant in places and just use the OpenAI role instead.

Reply:
FYI, this particular line is just like a default option here, so it'll still destructure the role from .delta if it exists.